h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = this.refs.searchBarInput;
+    this._dropdown = this.refs.searchBarDropdown;
+    this._dropdownItems = element.querySelectorAll('[data-ref="searchBarDropdownItem"]');
+
+    var getActiveElementIndex = () => {
```

Comment:
Is it really necessary to get the _index_ of the active element? Or could this be `getActiveElement()` instead, and then the code that uses it could work with the element itself rather than the element's index?

For example `updateActiveDropdownItem(index)` could become `updateActiveDropdownItem(element)` (where `element` is the element that should become the new active element). 

The code in `setupListenerKeys()` that gets the active element's index, increments it, and passes the new index to `updateActiveDropdownItem(index)` could instead get the active element, find its next sibling (`element.nextSibling` I think), and pass the next sibling element to `updateActiveDropdownItem(element)`.

I think this should allow the use of an `index` variable to be completely removed.